drivers/platform/aducm3029/gpio.c

```diff
@@ -0,0 +1,236 @@
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/************************* Include Files **************************************/
+/******************************************************************************/
+
+#include "error.h"
+#include "gpio.h"
+#include <drivers/gpio/adi_gpio.h>
+#include <stdlib.h>
+
+/******************************************************************************/
+/********************** Macros and Constants Definitions **********************/
+/******************************************************************************/
+
+/** Get GPIO pin from GPIO number */
+#define PIN(nb)		(1u << ((nb) & 0x0F))
+/** Get GPIO port from GPIO number */
+#define PORT(nb)	(((nb) & 0xF0) >> 4)
+
+/** Memory for GPIO device */
+static uint8_t mem_gpio_handler[ADI_GPIO_MEMORY_SIZE];
+/* Number of initialized devices */
+static uint8_t nb_gpio = 0;
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+
+/**
+ * @brief Obtain the GPIO descriptor from the number specified in param
+ * @param desc - Pointer to a structure were the descriptor will be stored.
+ * @param param - Parameter describing the GPIO to be initialized
+ * @return \ref SUCCESS in case of success, \ref FAILURE otherwise.
+ */
+int32_t gpio_get(struct gpio_desc **desc, const struct gpio_init_param *param)
+{
+	if (!desc || !param)
+		return FAILURE;
+
+	(*desc) = calloc(1, sizeof(**desc));
+	if (!(*desc))
+		return FAILURE;
+
+	(*desc)->number = param->number;
+
+	/* If this is the first GPIO initialize GPIO controller */
+	if(nb_gpio == 0)
```

Comment:
Add space after if, before bracket.

Reply:
Less spaces- > less memory for the source is needed.
Joke, solved :)